socketio/src/client/builder.rs

```diff
@@ -210,24 +209,48 @@ impl ClientBuilder {
 
     /// Uses a preconfigured TLS connector for secure communication. This configures
     /// both the `polling` as well as the `websocket` transport type.
-    /// # Example
-    /// ```rust
-    /// use rust_socketio::{ClientBuilder, Payload};
-    /// use native_tls::TlsConnector;
-    ///
-    /// let tls_connector =  TlsConnector::builder()
-    ///            .use_sni(true)
-    ///            .build()
-    ///            .expect("Found illegal configuration");
-    ///
-    /// let socket = ClientBuilder::new("http://localhost:4200/")
-    ///     .namespace("/admin")
-    ///     .on("error", |err, _| eprintln!("Error: {:#?}", err))
-    ///     .tls_config(tls_connector)
-    ///     .connect();
-    ///
-    /// ```
-    pub fn tls_config(mut self, tls_config: TlsConnector) -> Self {
+    #[cfg_attr(
+        feature = "_native-tls",
+        doc = r#"
+# Example for native-tls
```

Comment:
These look less appealing, but allows the docs build for all feature configurations